pkg/k8s/endpoints.go

```diff
@@ -357,7 +357,11 @@ func newExternalEndpoints() externalEndpoints {
 // SupportsEndpointSlice returns true if cilium-operator or cilium-agent should
 // watch and process endpoint slices.
 func SupportsEndpointSlice() bool {
-	return version.Capabilities().EndpointSlice && option.Config.K8sEnableK8sEndpointSlice
+	// MetalLB does not support EndpointSlices yet. See
+	// https://github.com/metallb/metallb/issues/811.
+	return version.Capabilities().EndpointSlice &&
+		option.Config.K8sEnableK8sEndpointSlice &&
+		!option.Config.BGPAnnounceLBIP
```

Comment:
This should be enough. If EndpointSlices don't replace Endpoints, but rather are an alternative, if available. In other words, if the apiserver supports them, but Cilium does **not** use them, then the Endpoint resource is still available for use. And that's what this code snippet is doing. 

However, I still think it's a good idea to add a warning as well, in case the user misses this call out in the docs. (See limitation section https://github.com/cilium/cilium/pull/15519/files#diff-9c888dd144df5108aaa6e5594733faf9522a81d7b6622c39c1cb973ff4a1644aR144)

Btw, I decided to drop this commit entirely actually and and move to disabling logic to simply toggling `option.Config.K8sEnableK8sEndpointSlice` while also logging the warning at the same time. That renders this commit useless, hence dropped.